src/extended-repeater.js

```diff
@@ -1,7 +1,21 @@
 const CustomError = require("../extensions/custom-error");
 
-module.exports = function repeater(/* str, options */) {
-  throw new CustomError('Not implemented');
-  // remove line with error and write your code here
-};
+module.exports = function repeater(repeatString, repeatOptions) {
+  let separator = repeatOptions['separator'] || '+';
+  let additionSeparator = repeatOptions['additionSeparator'] || '|';
+  let addition = repeatOptions['addition'];
+  let additionString;
+  try {
+    repeatString = repeatString.toString();
+    addition = addition.toString();
+  }
+  catch (error) {}
+  if (typeof(addition) !== "undefined")
```

Comment:
undefined transforms to false

if (!addition) { ....

Reply:
It was not an ideal code, I rewrote it. Please review again